pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -18,96 +18,92 @@ type ServiceBindingRequestSpec struct {
 	// +optional
 	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
 
-	// Custom env variables
+	// DataMapping is used to map CR/CSV/CRD attributes to Custom env variables
 	// +optional
-	CustomEnvVar []corev1.EnvVar `json:"customEnvVar,omitempty"`
-
-	// BackingServiceSelector is used to identify the backing service operator.
-	// Deprecation Notice:
-	// In the upcoming release, this field would be depcreated. It would be mandatory
-	// to set "backingServiceSelectors".
-	// +optional
-	BackingServiceSelector *BackingServiceSelector `json:"backingServiceSelector,omitempty"`
+	DataMapping []corev1.EnvVar `json:"dataMapping,omitempty"`
 
 	// BackingServiceSelectors is used to identify multiple backing services.
-	// This would be made a required field after 'BackingServiceSelector'
+	// This would be made a required field after 'services'
 	// is removed.
 	// +optional
-	BackingServiceSelectors *[]BackingServiceSelector `json:"backingServiceSelectors,omitempty"`
+	Services []Service `json:"services,omitempty"`
 
-	// ApplicationSelector is used to identify the application connecting to the
+	// Application is used to identify the application connecting to the
 	// backing service operator.
 	// +optional
-	ApplicationSelector ApplicationSelector `json:"applicationSelector"`
+	Application Application `json:"application"`
 
 	// DetectBindingResources is flag used to bind all non-bindable variables from
 	// different subresources owned by backing operator CR.
 	// +optional
 	DetectBindingResources bool `json:"detectBindingResources,omitempty"`
 }
 
-// ServiceBindingRequestStatus defines the observed state of ServiceBindingRequest
+// ServiceBindingStatus defines the observed state of ServiceBindingStatus
 // +k8s:openapi-gen=true
-type ServiceBindingRequestStatus struct {
+type ServiceBindingStatus struct {
 	// Conditions describes the state of the operator's reconciliation functionality.
 	Conditions []conditionsv1.Condition `json:"conditions"`
 	// Secret is the name of the intermediate secret
-	Secret string `json:"secret"`
+	Secret corev1.LocalObjectReference `json:"secret"`
 	// ApplicationObjects contains all the application objects filtered by label
 	// +optional
 	Applications []BoundApplication `json:"applications,omitempty"`
 }
 
-// BackingServiceSelector defines the selector based on resource name, version, and resource kind
-type BackingServiceSelector struct {
-	metav1.GroupVersionKind `json:",inline"`
-	ResourceRef             string `json:"resourceRef"`
+// Service defines the selector based on resource name, version, and resource kind
+type Service struct {
+	metav1.GroupVersionKind     `json:",inline"`
+	corev1.LocalObjectReference `json:",inline"`
```

Comment:
We have `LabelSelector` for Service.
Let's have it for `Services` as well. We do support multiple services.

Reply:
Wondering if that might the way we do:
* referring to backing services in custom env var.
* using env prefixes per service.